qiskit_ibm_transpiler/ai/synthesis.py

```diff
@@ -383,7 +381,7 @@ def __init__(
         local_mode: bool = False,
         **kwargs,
     ) -> None:
-        if local_mode == True:
+        if local_mode == True:  # noqa:E712
```

Comment:
I guess that here we want to check also that local_mode is a boolean, just in case `local_mode` is another type (a string with value "false", for example) and Python can process it as a truthy value.
Maybe adding something like:
```
 if isinstance(local_mode, bool) and local_mode:
```
could work

Reply:
I see, we could do it but in the parent class we only check `if local_mode:` and we are specifying in the arguments that it should be a boolean, so I won't check if it is a boolean. You can leave the `# noqa` if you prefer.